lms/djangoapps/course_goals/api.py

```diff
@@ -21,11 +23,21 @@ def add_course_goal(user, course_id, goal_key):
         user: The user that is setting the goal
         course_id (string): The id for the course the goal refers to
         goal_key (string): The goal key for the new goal.
+        number_of_days_with_visits_per_week_goal (int): number of days learner wants to learn per week
+        subscribed_to_goal_reminders (bool): whether the learner wants to receive email reminders about their goal
     """
     course_key = CourseKey.from_string(str(course_id))
-    CourseGoal.objects.update_or_create(
-        user=user, course_key=course_key, defaults={'goal_key': goal_key}
-    )
+    if goal_key:
```

Comment:
This feels like two different methods - `set_course_goal_as_key` and `set_course_goal_as_visits` or something. Or just `set_course_goal_legacy` and `set_course_goal`. Feels weird to implicitly have 3 optional arguments to this method, and leave the headache of knowing what is what to the callers of this method: callers have to know (a) which combination of them go together and (b) that they actively shouldn't pass all three if they have them, because that would lead to two of them being ignored, etc.